src/aero_reps/aero_rep_single_particle_layers.F90

```diff
@@ -0,0 +1,754 @@
+  !! provided to \c aero_rep_single_particle::initialize()
+  function get_size(this) result (state_size)
+
+    !> Size on the state array
+    integer(kind=i_kind) :: state_size
+    !> Aerosol representation data
+    class(aero_rep_single_particle_t), intent(in) :: this
+
+    state_size = MAX_PARTICLES_ * PARTICLE_STATE_SIZE_
+
+  end function get_size
+
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+  !> Get the number of state variables per-particle
+  !!
+  !! Calling functions can assume each particle has the same size on the
+  !! state array, and that individual particle states are contiguous and
+  !! arranged sequentially
+  function per_particle_size(this) result(state_size)
+
+    !> Size on the state array per particle
+    integer(kind=i_kind) :: state_size
+    !> Aerosol representation data
+    class(aero_rep_single_particle_t), intent(in) :: this
+
+    state_size = PARTICLE_STATE_SIZE_
+
+  end function per_particle_size
+
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+  !> Get a list of unique names for each element on the
+  !! \c camp_camp_state::camp_state_t::state_var array for this aerosol
+  !! representation. The list may be restricted to a particular phase and/or
+  !! aerosol species by including the phase_name and spec_name arguments.
+  !!
+  !! For a single particle representation, the unique names will be a 'P'
+  !! followed by the computational particle number, a '.', the layer name,
+  !! another '.', the phase name, another '.', and the species name.
+  function unique_names(this, layer_number, phase_name, tracer_type, spec_name)
+
+    use camp_util,                      only : integer_to_string
+
+    !> List of unique names
+    type(string_t), allocatable :: unique_names(:)
+    !> Aerosol representation data
+    class(aero_rep_single_particle_t), intent(in) :: this
+    !> Aerosol layer name
+    character(len=*), optional, intent(in) :: layer_name
+    !> Aerosol phase name
+    character(len=*), optional, intent(in) :: phase_name
+    !> Aerosol-phase species tracer type
+    integer(kind=i_kind), optional, intent(in) :: tracer_type
+    !> Aerosol-phase species name
+    character(len=*), optional, intent(in) :: spec_name
+
+    integer(kind=i_kind) :: num_spec, i_part, i_spec, j_spec, i_phase
+    integer(kind=i_kind) :: num_layer, i_layer
+    integer(kind=i_kind) :: curr_tracer_type
+    character(len=:), allocatable :: curr_phase_name
+    character(len=:), allocatable :: curr_layer_name
+    type(string_t), allocatable :: spec_names(:)
+
+    ! Copy saved unique names when no filters are included
+    if (.not. present(layer_name) .and. &
+        .not. present(phase_name) .and. &
+        .not. present(tracer_type) .and. &
+        .not. present(spec_name) .and. &
+        allocated(this%unique_names_)) then
+      unique_names = this%unique_names_
+      return
+    end if
+
+    ! Count the number of unique names
```

Comment:
lines of code containing phase were copied and replaced with layer.  is this correct?